adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -686,6 +688,65 @@ private void verifyBrokerRedirectUri(final AuthenticationRequest request) throws
         Logger.v(TAG + methodName, "The broker redirect URI is valid.");
     }
 
+    private AuthenticationResult getAuthenticationResultFromIntent(final Intent data) {
+        final String methodName = ":getAuthenticationResultFromIntent";
+
+        final String accessToken = data
+                .getStringExtra(AuthenticationConstants.Broker.ACCOUNT_ACCESS_TOKEN);
+        final String accountName = data
+                .getStringExtra(AuthenticationConstants.Broker.ACCOUNT_NAME);
+        mBrokerProxy.saveAccount(accountName);
+        final long expireTime = data.getLongExtra(
+                AuthenticationConstants.Broker.ACCOUNT_EXPIREDATE, 0);
+        final Date expire = new Date(expireTime);
+        final String idtoken = data.getStringExtra(AuthenticationConstants.Broker.ACCOUNT_IDTOKEN);
+        final String tenantId = data.getStringExtra(
+                AuthenticationConstants.Broker.ACCOUNT_USERINFO_TENANTID);
+        UserInfo userinfo = null;
+        if (null != data.getExtras()) {
+            userinfo = UserInfo.getUserInfoFromBrokerResult(data.getExtras());
+        }
+
+        // grab the fields tracked by x-ms-clitelem
+        final String serverErrorCode = data.getStringExtra(AuthenticationConstants.Broker.CliTelemInfo.SERVER_ERROR);
+        final String serverSubErrorCode = data.getStringExtra(AuthenticationConstants.Broker.CliTelemInfo.SERVER_SUBERROR);
+        final String refreshTokenAge = data.getStringExtra(AuthenticationConstants.Broker.CliTelemInfo.RT_AGE);
+        final String speRingInfo = data.getStringExtra(AuthenticationConstants.Broker.CliTelemInfo.SPE_RING);
+//
+//        // Use the waitingRequest to get the original request to get the clientId
+//        final AuthenticationRequest originalRequest = waitingRequest.getRequest();
+//        String clientId = null;
+//
+//        if (null != originalRequest) {
+//            clientId = originalRequest.getClientId();
+//        }
+
+        // create the broker AuthenticationResult
+        final AuthenticationResult brokerResult =
+                new AuthenticationResult(
+                        accessToken,
+                        null,
+                        expire,
+                        false,
+                        userinfo,
+                        tenantId,
+                        idtoken,
+                        null,
+                        ""
```

Comment:
Does the `clientId` get set somewhere else? Why `""` and not `null`?

Also, it looks like the `clientId` may be available from `waitingRequest.getRequest().getClientId()` --  if that value is populated, shouldn't it passed into this method such that it can be used to construct the `AuthenticationResult`?